src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -112,6 +112,178 @@ public class DerivedOptionsWithIConfigurationSection : DerivedOptions
+                String2 = string2;
+                Int2 = int2;
+            }
+
+            public string String1 { get; }
+            public string String2 { get; }
+            public int Int1 { get; }
+            public int Int2 { get; }
+        }
+
+        public class ImmutableClassWithOneParameterisedConstructorButWithRefParameter
+        {
+            public ImmutableClassWithOneParameterisedConstructorButWithRefParameter(string string1, ref int int1, string string2, int int2)
+            {
+                String1 = string1;
+                Int1 = int1;
+                String2 = string2;
+                Int2 = int2;
+            }
+
+            public string String1 { get; }
+            public string String2 { get; }
+            public int Int1 { get; }
+            public int Int2 { get; }
+        }
+
+        public class ImmutableClassWithOneParameterisedConstructorButWithOutParameter
+        {
+            public ImmutableClassWithOneParameterisedConstructorButWithOutParameter(string string1, int int1,
+                string string2, out decimal int2)
+            {
+                String1 = string1;
+                Int1 = int1;
+                String2 = string2;
+                int2 = 0;
+            }
+
+            public string String1 { get; }
+            public string String2 { get; }
+            public int Int1 { get; }
+            public int Int2 { get; }
+        }
+
+        public class ImmutableClassWithMultipleParameterisedConstructors
+        {
+            public ImmutableClassWithMultipleParameterisedConstructors(string string1, int int1)
+            {
+                String1 = string1;
+                Int1 = int1;
+            }
+
+            public ImmutableClassWithMultipleParameterisedConstructors(string string1, int int1, string string2)
+            {
+                String1 = string1;
+                Int1 = int1;
+                String2 = string2;
+            }
+
+            public ImmutableClassWithMultipleParameterisedConstructors(string string1, int int1, string string2, int int2)
+            {
+                String1 = string1;
+                Int1 = int1;
+                String2 = string2;
+                Int2 = int2;
+            }
+
+            public ImmutableClassWithMultipleParameterisedConstructors(string string1)
+            {
+                String1 = string1;
+            }
+
+            public string String1 { get; }
+            public string String2 { get; }
+            public int Int1 { get; }
+            public int Int2 { get; }
+        }
+
+        public class SemiImmutableClass
+        {
+            public SemiImmutableClass(string color, int length)
+            {
+                Color = color;
+                Length = length;
+            }
+
+            public string Color { get; }
+            public int Length { get; }
+            public decimal Thickness { get; set; }
+        }
+
+        public class SemiImmutableClassWithInit
```

Comment:
What's the difference between this class and `SemiImmutableClass`? Was this supposed to have 

```C#
public decimal Thickness { get; init; }
```

?

Reply:
Yes, exactly that - well spotted, thanks!